File: Python/Practice/programmers_092.py
def solution(n):
    answer = 0
    for i in range(1, n):
        num = i
        snum = 0
        while snum < n:
            snum += num
            num += 1
        if snum == n:
            answer += 1
    return answer + 1

# 공식 적용
def solution(num):
    return len([i for i in range(1, num+1, 2) if num % i is 0])

File: JavaScript/programmers_092.js
// LV.2 [1차] 캐시

function solution(cacheSize, cities) {
  let answer = 0;
  let arr = [];
  let len = cities.length;
  if (cacheSize === 0) {
    return len * 5;
  }
  for (let i = 0; i < len; i++) {
    if (arr.includes(cities[i].toLowerCase())) {
      arr.splice(arr.indexOf(cities[i].toLowerCase()), 1);
      arr.push(cities[i].toLowerCase());
      answer += 1;
    } else {
      if (arr.length >= cacheSize) arr.shift();
      arr.push(cities[i].toLowerCase());
      answer += 5;
    }
  }
  return answer;
}
